src/AWS.Deploy.Orchestration/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,264 @@
+        /// <param name="saveCdkDirectoryFullPath">The absolute path to the directory at which the CDK deployment project is saved</param>
+        /// <exception cref="FailedToUpdateDeploymentManifestFileException">Thrown if an error occured while trying to update the deployment manifest file.</exception>
+        /// </summary>
+        public async Task UpdateSaveCdkProject(string saveCdkDirectoryFullPath)
+        {
+            try
+            {
+                if (!_directoryManager.Exists(saveCdkDirectoryFullPath))
+                    return;
+
+                var deploymentManifestModel = await GetDeploymentManifest();
+                var targetApplicationDirectoryPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+                var saveCdkDirectoryRelativePath = _directoryManager.GetRelativePath(targetApplicationDirectoryPath, saveCdkDirectoryFullPath);
+
+                if (deploymentManifestModel != null && deploymentManifestModel.DeploymentManifestEntries != null)
+                {
+                    deploymentManifestModel.DeploymentManifestEntries.Add(new DeploymentManifestEntry(saveCdkDirectoryRelativePath));
+                }
+                else
+                {
+                    var deploymentManifestEntries = new List<DeploymentManifestEntry> { new DeploymentManifestEntry(saveCdkDirectoryRelativePath) };
+                    deploymentManifestModel = new DeploymentManifestModel(new(), deploymentManifestEntries);
+                }
+
+                await WriteDeploymentManifestFile(deploymentManifestModel);
+            }
+            catch (Exception ex)
+            {
+                throw new FailedToUpdateDeploymentManifestFileException($"Failed to update the deployment manifest file " +
+                    $"for the deployment project stored at '{saveCdkDirectoryFullPath}'", ex);
+            }
+            
+        }
+
+        /// <summary>
+        /// This method updates the deployment manifest json file by adding the name of the stack that was most recently used.
+        /// If the manifest file does not exists then a new file is generated.
+        /// </summary>
+        public async Task UpdateLastDeployedStack(string stackName)
+        {
+            try
+            {
+                if (_session.AWSAccountId == null || _session.AWSRegion == null)
+                    throw new FailedToUpdateDeploymentManifestFileException("The AWS Account Id or Region is not defined.");
+
+                var deploymentManifestModel = await GetDeploymentManifest();
+                var lastDeployedStack = deploymentManifestModel?.LastDeployedStacks?
+                    .FirstOrDefault(x => x.AWSAccountId.Equals(_session.AWSAccountId) && x.AWSRegion.Equals(_session.AWSRegion));
+
+                if (deploymentManifestModel != null && lastDeployedStack != null)
+                {
+                    if (lastDeployedStack != null)
```

Comment:
Since we are sorting by the CloudFormation update timestamp we don't need to keep changing the order of this file. I think that will cause frustration with a pair of developers both deploying to their own stacks in the same account but the file keeps getting changes and then being listed as a changed file for change control. If we wanted to have a deterministic  list just always keep the list in alpha order.